### ISSUE TITLE
DITA Bootstrap + Lunr Search - How to document it?

### ISSUE THREAD

**Issue by jason-fox** [infotexture/dita-bootstrap#75]
I have created an extension to **DITA Bootstrap** for a search box which incorporates a [lunr.js](https://lunrjs.com/) search function. You can see the source code [here](https://github.com/jason-fox/dita-bootstrap.lunr) and examples of the result can be seen [here](https://jason-fox.github.io/dita-bootstrap/) and [here](https://jason-fox.github.io/dita-ot-plugins/)

I'm now wondering what to do next. 🤷 

<img width="808" alt="Screenshot 2023-04-12 at 19 27 07" src="https://user-images.githubusercontent.com/3439249/231537898-d624b2b6-8065-4eb4-8942-30616bb21ff3.png">

### Thoughts

Although the generation of the lunr search index is generic, the display of the results is currently very tightly bound to the structure of the HTML created by the **DITA Bootstrap** plugin, and I'm wondering how to incorporate this in a coherent manner.

1.  **Lunr.js search** shouldn't be part of the main  **DITA Bootstrap** plugin - there are plenty of other sensible options - e.g. Solr, Algolia, Google etc. - this is just one way of doing it.
2.  Since **Lunr.js search**  is so tightly bound, the repo would be better placed somewhere under your account - i.e. `infotexture/dita-bootstrap.lunr` rather  mine `jason-fox/dita-bootstrap.lunr` - it would be more discoverable that way.
3.  For Documentation, it would be possible to add in another topic in the   **DITA Bootstrap** docs under *Extend* like *Icons* or *Dark Mode*
4. The CI generated GitHub Action can be altered to add search of course.

### Questions

So the question is what should happen next? Would it lie better under your account or mine? Should I cede it over to you and you give me back collaborator access or should I leave it where it is? How should I go about raising a documentation PR?

What are your thoughts?

**Comment by infotexture** [comment]
@jason-fox Incredibly timely premonition on your part! 🔮 👏  

For a few weeks now, I've had an item on my ToDo-list to file a new issue in this repo to add local search to the DITA Bootstrap plug-in, and planned to solicit your input on the best approach. Thanks for beating me to it. 😀 

I was thinking in the same directions:

- The search implementation would depend on the search service, which would justify a separate plug-in that extends DITA Bootstrap, rather than adding it directly, as not everyone would want/need a (particular) search implementation
- The rendering of results would require the Bootstrap HTML/CSS, so search plug-in(s) would require DITA Bootstrap, instead of being standalone plug-ins
- Lunr seems like a natural starting point as a lightweight open-source implementation
- Another search option would be https://pagefind.app, but that could be separate

As for which account to publish & maintain it from, I have no strong preference.

- Since you've already built it, I wouldn't want to take credit for your work
- If you feel it belongs under `infotexture/dita-bootstrap.lunr` alongside this plug-in, it would be very welcome here
- Either way, I agree we could document it here under _Extend_, where we can explain how to install the additional plug-in and configure the build/actions to re-generate the search index

**Comment by jason-fox** [comment]
Sounds like a plan. Let's move lunr search to `infotexture/dita-bootstrap.lunr`, but keep the copyright as me  😄 (rather than you 😞 )  on the` README` - You make me a collaborator and I can refork back for further work. Release 0.0.1 of `infotexture/dita-bootstrap.lunr` can be added to the Plug-ins repo, so that the _"install and how to use"_ can be added to the  **DITA Bootstrap** docs, which should drop a release in the near future as Bootstrap 5.3 is imminent.

Release 1.0.0 of `infotexture/dita-bootstrap.lunr` should occur once the internationalization is complete - at the moment it is  English only.

**Comment by infotexture** [comment]
@jason-fox Cloned jason-fox/dita-bootstrap.lunr@ec0bb00 to [infotexture/dita-bootstrap.lunr](https://github.com/infotexture/dita-bootstrap.lunr) & added you as collaborator.

I like to use Git Flow, so I set the default branch to `develop` for ongoing work.
